test/realm_test.dart

```diff
@@ -1756,6 +1755,125 @@ Future<void> main([List<String>? args]) async {
+    bool isRefreshed = false;
+    final transaction = realm.beginWrite();
+    realm.refreshAsync().then((refreshed) {
+      called = true;
+      isRefreshed = refreshed;
+    });
+    realm.add(Person("name"));
+    transaction.commit();
+
+    await Future<void>.delayed(Duration(milliseconds: 1));
+    expect(isRefreshed, false, reason: "There are no new data");
+    expect(called, true, reason: "Refresh called after transaction commit");
+    expect(realm.all<Person>().length, 1);
+  });
+
+  test('Realm.refresh into a write block', () async {
+    final realm = getRealm(Configuration.local([Person.schema]));
+    var called = false;
+    bool isRefreshed = false;
+    realm.write(() {
+      realm.refreshAsync().then((refreshed) {
+        called = true;
+        isRefreshed = refreshed;
+      });
+      realm.add(Person("name"));
+    });
+
+    await Future<void>.delayed(Duration(milliseconds: 1));
+    expect(isRefreshed, false, reason: "There are no new data");
+    expect(called, true, reason: "Refresh called after write block completes");
+    expect(realm.all<Person>().length, 1);
+  });
+
+  test('Realm.refresh into async transaction', () async {
+    final realm = getRealm(Configuration.local([Person.schema]));
+    bool called = false;
+    bool isRefreshed = false;
+    final transaction = await realm.beginWriteAsync();
+    realm.refreshAsync().then((refreshed) {
+      called = true;
+      isRefreshed = refreshed;
+    });
+    realm.add(Person("name"));
+    await transaction.commitAsync();
+    expect(isRefreshed, false, reason: "There are no new data");
+    expect(called, true, reason: "Refresh called after transaction commit");
+    expect(realm.all<Person>().length, 1);
+  });
+
+  test('Realm.refresh is called if registered outside a transaction', () async {
+    final realm = getRealm(Configuration.local([Person.schema]));
+    bool called = false;
+    bool isRefreshed = false;
+   
+    realm.refreshAsync().then((refreshed) {
+      called = true;
+      isRefreshed = refreshed;
+    });
+
+    realm.write(() {});
+    await Future<void>.delayed(Duration(milliseconds: 1));
+
+    expect(isRefreshed, false, reason: "There are no new data");
+    expect(called, true);
+  });
+
+  test('Realm.refresh on frozen realm should be no-op', () async {
+    var realm = getRealm(Configuration.local([Person.schema]));
+    bool called = false;
+    realm = realm.freeze();
+    final isRefreshed = await realm.refreshAsync();
+    expect(isRefreshed, false);
+  });
+
+  test('Realm.refresh is updating data', () async {
+    Completer<void> completer = Completer<void>();
+    final realm = getRealm(Configuration.local([Person.schema]));
+    String personName = generateRandomString(5);
+    final path = realm.config.path;
+    final results = realm.query<Person>(r"name == $0", [personName]);
+
+    expect(realm.refresh(), false);
+    realm.setAutoRefresh(false);
+
+    ReceivePort receivePort = ReceivePort();
```

Comment:
This test seems unnecessary complex. Why are we sending booleans and waiting for 100 mils?

Reply:
I would expect this to work as well

```dart
 final externalRealm = Realm(Configuration.local([Person.schema], path: path));
      externalRealm.write(() {
        externalRealm.add(Person(personName));
      });
    externalRealm.close();
```

Reply:
You suggested to close the second realm after the refresh was called. That's why I close it a little bit later. If you think it is not need I can simplify the test.

Reply:
I think it should not be needed if we disable autorefresh. My suggestion was how to workaround autorefresh. Which is not needed now

Reply:
Done.